tests/functional/conftest.py

```diff
@@ -48,10 +51,40 @@
 # Directory storing test-specific data.
 _DATA_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data')
 
-# Provide test access to _DATA_DIR via a fixture.
+# Code
+# ====
+# Fixtures
+# --------
+# This class provides access to the _DATA_DIR, and also copies correctly-named
+# directories in the _DATA_DIR to the tmpdir.
+class DataDir(object):
+    # Optionally copy from _DATA_DIR to tmpdir.
+    def __init__(self,
+      # The request object for this test. See
+      # https://pytest.org/latest/builtin.html#_pytest.python.FixtureRequest
+      # and
+      # https://pytest.org/latest/fixture.html#fixtures-can-introspect-the-requesting-test-context.
+      request,
+      # The tmpdir object for this test. See
+      # https://pytest.org/latest/tmpdir.html.
+      tmpdir):
+
+        # Provide access to _DATA_DIR, using the same nomenclature as tmpdir.
+        self.strpath = _DATA_DIR
```

Comment:
`tmpdir` has a `strpath` because it is a `py.path.local` object. I don't think this fact is well-exploited in the existing test suites.